File: src/Orchestrator/src/Application/General/Queries/GetContractsStates/VM/SettlementProposalVm.cs
namespace Application.General.Queries.GetContractsStates.VM;

public class SettlementProposalVm
{
    public required Guid Id { get; init; }
    public required string Title { get; init; }
    public SettlementProposalAssessmentVerifierLotteryVm? Lottery { get; init; }
    public PollVm? Poll { get; init; }
}

File: src/frontend/lib/admin/models/vm/settlement_proposal_vm.dart
import 'poll_vm.dart';
import 'settlement_proposal_assessment_verifier_lottery_vm.dart';

class SettlementProposalVm {
  final String id;
  final String title;
  final SettlementProposalAssessmentVerifierLotteryVm? lottery;
  final PollVm? poll;

  SettlementProposalVm.fromMap(Map<String, dynamic> map)
      : id = map['id'],
        title = map['title'],
        lottery = map['lottery'] != null ? SettlementProposalAssessmentVerifierLotteryVm.fromMap(map['lottery']) : null,
        poll = map['poll'] != null ? PollVm.fromMap(map['poll']) : null;
}
